### PR TITLE
Introduce SERVICES_HOST environment variable

### LINKED ISSUE #97: A way to tell the plugin to look at localhost for health checks.
Hi,

I've been using your plugin for a while, very successfully on CircleCI 1.0

However, we're now moving to CircleCI 2.0 which doesn't allow you to run your own docker instance - instead, it runs a remote one for you and when you compose containers, they run on a remote host.

The problem is, the build itself on CircleCI also runs in a docker container, and that container cannot directly reach the remote docker host. What happens is that CircleCI binds all exposed ports to the build container.

As a result my healthcheck output is as follows:

```
> Building 50% > :composeUpProbing TCP socket on 35.190.191.100:3002 of service 'api'
Waiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
> Building 50% > :composeUpWaiting for TCP socket on 35.190.191.100:3002 of service 'api' (Connection timed out (Connection timed out))
```

So the plugin correctly detects the remote host, but isn't aware that CircleCI has made the service TCP ports accessible on 'localhost' instead.

Looking at the source, I can't see a way to tell the plugin:

`detect the remote docker instance, but expect the ports to be bound on xxx.yyy (or localhost)`

I'm wondering about turning the healthcheck off and doing it manually for now, but it's not ideal.

### PR DESCRIPTION
**Pull request by augi**
Fixes #97

### CODE REVIEW

**Comment by antony** [comment]
Hi @augi 

I think it half works, but you aren't explicitly returning the new ServiceHost from your method, so it is being created and dropping through into the 'else' of the following block.

As a result it assigns it, and then immediately discards it, reverting to using the network bridge (this is local, so it is actually accessible here - but it should be probing on localhost, and exposing localhost as the final hostname for services)

```javascript
'SERVICES_HOST environment variable detected - will be used as hostname of service mongo (localhost)'
Will use 172.17.0.1 (network bridge) as host of mongo
'SERVICES_HOST environment variable detected - will be used as hostname of service api (localhost)'
Will use 172.17.0.1 (network bridge) as host of api
'SERVICES_HOST environment variable detected - will be used as hostname of service client (localhost)'
Will use 172.17.0.1 (network bridge) as host of client
Probing TCP socket on 172.17.0.1:3002 of service 'api_1'
TCP socket on 172.17.0.1:3002 of service 'api_1' is ready
Probing TCP socket on 172.17.0.1:80 of service 'client_1'
TCP socket on 172.17.0.1:80 of service 'client_1' is ready
```

File: src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy [hunk 1]
```diff
@@ -212,6 +212,11 @@ class ComposeUp extends DefaultTask {
     }
 
     ServiceHost getServiceHost(String serviceName, Map<String, Object> inspection) {
+        String servicesHost = extension.environment['SERVICES_HOST'] ?: System.getenv('SERVICES_HOST')
+        if (servicesHost) {
+            logger.lifecycle("'SERVICES_HOST environment variable detected - will be used as hostname of service $serviceName ($servicesHost)'")
+            new ServiceHost(host: servicesHost, type: ServiceHostType.RemoteDockerHost)
```
Comment (antony) [inline src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: probably just explictly return this for it work:

```groovy
             return new ServiceHost(host: servicesHost, type: ServiceHostType.RemoteDockerHost)
```

Comment (antony) [reply src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: also the logging comes out with quotes. You can probably drop the ' and ' within your logger line.

Comment (augi) [reply src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy]: Fixed.

**Comment by augi** [comment]
I've fixed both issues.

**Comment by antony** [comment]
Thanks - I didn't expect you to resolve this so quickly so I tried out my suggestions myself. I've opened another PR here: https://github.com/avast/gradle-docker-compose-plugin/pull/99

I also refactored a bit because the method was getting quite large.

Both of the PRs work, so take what you prefer.

Thanks,
Antony

**Pull request merged**